pkg/scheduler/apis/config/v1alpha1/defaults.go

```diff
@@ -78,11 +78,11 @@ func SetDefaults_KubeSchedulerConfiguration(obj *kubescedulerconfigv1alpha1.Kube
 		obj.MetricsBindAddress = net.JoinHostPort("0.0.0.0", strconv.Itoa(ports.InsecureSchedulerPort))
 	}
 
-	if len(obj.LeaderElection.LockObjectNamespace) == 0 {
-		obj.LeaderElection.LockObjectNamespace = kubescedulerconfigv1alpha1.SchedulerDefaultLockObjectNamespace
+	if len(obj.LeaderElection.LockObjectNamespace) == 0 && len(obj.LeaderElection.ResourceNamespace) == 0 {
+		obj.LeaderElection.ResourceNamespace = kubeschedulerconfigv1alpha1.SchedulerDefaultLockObjectNamespace
 	}
-	if len(obj.LeaderElection.LockObjectName) == 0 {
-		obj.LeaderElection.LockObjectName = kubescedulerconfigv1alpha1.SchedulerDefaultLockObjectName
+	if len(obj.LeaderElection.LockObjectName) == 0 && len(obj.LeaderElection.ResourceName) == 0 {
+		obj.LeaderElection.ResourceName = kubeschedulerconfigv1alpha1.SchedulerDefaultLockObjectName
```

Comment:
keep defaulting setting the LockObjectName field. we want the files we generate to work for n-1 schedulers that only read that field